src/entry.c

```diff
@@ -0,0 +1,385 @@
+
+    bool expiry_add_remove = update_expiry && (curr_expiration_time == EXPIRY_NONE || expiry == EXPIRY_NONE); // In case we are toggling expiration
+    bool value_change_encoding = update_value && (embed_value != !entryHasValuePtr(e));                       // In case we change the way value is embedded or not
+
+
+    /* // We will create a new entry in the following cases:
+     * 1. In the case were we add or remove expiration.
+     * 2. We change the way value is encoded
+     * 3. in the case were we are NOT migrating from an embedded entry to an embedded entry with ~the same size. */
+    bool create_new_entry = (expiry_add_remove) || (value_change_encoding) ||
+                            (update_value && !entryHasValuePtr(e) &&
+                             !(required_embedded_size <= EMBED_VALUE_MAX_ALLOC_SIZE &&
+                               required_embedded_size <= current_embedded_allocation_size &&
+                               required_embedded_size >= current_embedded_allocation_size * 3 / 4));
+
+    if (!create_new_entry) {
+        /* In this case we are sure we do not have to allocate new entry, so expiry must already be set. */
+        if (update_expiry) {
+            serverAssert(entryHasExpiry(e));
+            char *buf = entryGetAllocPtr(e);
+            *(long long *)buf = expiry;
+        }
+        /* In this case we are sure we do not have to allocate new entry, so value must already be set or we have enough room to embed it. */
+        if (update_value) {
+            if (entryHasValuePtr(e)) {
+                sds *value_ref = entryGetValueRef(e);
+                sdsfree(*value_ref);
+                *value_ref = value;
+            } else {
+                /* Skip field content, field null terminator and value sds8 hdr. */
+                sds old_value = entryGetValue(e);
+                /* We are using the same entry memory in order to store a potentially new value.
+                 * In such cases the old value alloc was adjusted to the real buffer size part it was embedded to.
+                 * Since we can potentially write here a smaller value, which requires less allocation space, we would like to
+                 * inherit the old value memory allocation size. */
+                size_t value_size = sdsHdrSize(SDS_TYPE_8) + sdsalloc(old_value) + 1;
+                sdswrite(sdsAllocPtr(old_value), value_size, SDS_TYPE_8, value, sdslen(value));
+                sdsfree(value);
+            }
+        }
+        new_entry = e;
+
+    } else {
+        if (!update_value) {
+            /* Check if the value can be reused. */
+            int value_was_embedded = !entryHasValuePtr(e);
+            /* In case the original entry value is embedded WE WILL HAVE TO DUPLICATE IT */
+            if (value_was_embedded)
+                value = sdsdup(value);
+            /* if not we have to duplicate it, remove it from the original entry since we are going to delete it.*/
```

Comment:
iirc, we'd like to have consistent style for both `if` and `else`

```suggestion
            if (value_was_embedded) {
                value = sdsdup(value);
             }
            /* if not we have to duplicate it, remove it from the original entry since we are going to delete it.*/
```

Reply:
sure. it did allow placing the comments in a more "aligned" way7. but I agree about the syntax